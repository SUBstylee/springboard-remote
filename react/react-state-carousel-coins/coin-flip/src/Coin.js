import heads from './img/heads.png';
import tails from './img/tails.png';

const Coin = (props) => {
    return <img src={props.info === 'heads' ? heads : tails} alt={props.info} />;
};

export default Coin;