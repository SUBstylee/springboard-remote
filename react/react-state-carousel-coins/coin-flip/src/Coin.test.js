import Coin from "./Coin";
import { render } from "@testing-library/react";


it('renders without crashing', () => {
    render(<Coin info='heads' />)

});

it('matches snapshot', () => {
    const { asFragment } = render(<Coin />);
    expect(asFragment()).toMatchSnapshot();
});

it('shows heads', () => {
    const { queryByAltText } = render(<Coin info='heads' />);
    expect(queryByAltText('heads')).toBeInTheDocument();
    expect(queryByAltText('tails')).not.toBeInTheDocument();
});

it('shows tails', () => {
    const { queryByAltText } = render(<Coin info='tails' />);
    expect(queryByAltText('tails')).toBeInTheDocument();
    expect(queryByAltText('heads')).not.toBeInTheDocument();
});