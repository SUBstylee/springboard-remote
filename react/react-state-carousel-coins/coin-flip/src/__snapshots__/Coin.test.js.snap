// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <img
    src="tails.png"
  />
</DocumentFragment>
`;
